tests/foreman/ui/test_host.py

```diff
@@ -1,14 +1,278 @@
 # -*- encoding: utf-8 -*-
-from nailgun import entities
+from fauxfactory import gen_string
+from nailgun import entities, entity_mixins
+from robottelo.api.utils import promote
+from robottelo.config import conf
 from robottelo.constants import ENVIRONMENT
 from robottelo.decorators import run_only_on
 from robottelo.test import UITestCase
+from robottelo.ui.base import Base
 from robottelo.ui.locators import common_locators
 from robottelo.ui.session import Session
 
 
 @run_only_on('sat')
-class Host(UITestCase):
+class Host(UITestCase, Base):
+
+    hostname = gen_string('numeric')
+
+    @classmethod
+    def setUpClass(cls):  # noqa
+        """Steps required to create a real host on libvirt
+
+        1. Creates new Organization and Location
+        2. Creates new life-cycle environment
+        3. Creates new product and rhel67 custom repository
+        4. Creates new content-view by associating rhel67 repository
+        5. Publish and promote the content-view to next environment
+        6. Search for puppet environment and associate location
+        7. Search for smart-proxy and associate location
+        8. Search for domain and associate org, location and dns proxy
+        9. Search for '192.168.100.0' network and associate org, location,
+        dns/dhcp/tftp proxy, and if its not there then creates new
+        10. Creates new compute-resource
+        11. Search 'Kickstart default' partition table and rhel67 OS along with
+        provisioning/PXE templates
+        12. Associates org, location and OS with provisioning and PXE templates
+        13. Search for x86_64 architecture
+        14. Associate arch, partition table, provisioning/PXE templates with OS
+        15. Search for media and associate org/location
+        16. Create new host group with all required entities
+
+        """
+        # Create a new Organization and Location
+        cls.org = entities.Organization(name=gen_string('alpha')).create()
```

Comment:
:+1: I also like to see only alpha numeric Org names since it is easy to search in my dropdown :)